cf-update-circl.py

```diff
@@ -0,0 +1,193 @@
+            bytes(circl_name, 'ascii'),
+        ).replace(
+            b'golang.org/x/sys/cpu',
+            b'internal/cpu',
+        )
+        with open(fn, 'wb') as f:
+            f.write(s)
+
+pkg_readme_template = '''
+THIS CODE WAS GENERATED BY cf-update-circl.py.
+
+It was checked out from "{0}"
+on branch "{1}". Latest commit:
+'''
+
+# Remove existing CIRCL package.
+if os.path.exists(src_circl):
+    print(f"Removing existing {src_circl}")
+    subprocess.check_call(['rm', '-r', '-f', src_circl])
+
+# Clone and copy new CIRCL package.
+with tempfile.TemporaryDirectory() as tmp:
+    pkg = {
+        os.path.join(*d.split('/')) : False for d in circl_pkg_dir
+    }
+
+    testdata = {
+        os.path.join(*d.split('/')) : False for d in circl_testdata_dir
+    }
+
+    misc = {
+        os.path.join(*f.split('/')) : False for f in circl_misc_file
+    }
+
+    print(f"Cloning {circl_repo} branch {circl_branch} ...")
+    subprocess.check_call(['git', 'clone',
+        circl_repo,
+        '--branch', circl_branch,
+        '--depth', '1',
+        circl_name,
+    ], cwd=tmp)
+
+    print(f"Creating new {src_circl} ...")
+    for subdir, _, files in os.walk(os.path.join(tmp, circl_name)):
+        d = os.path.relpath(subdir, tmp)
+
+        # Handle and copy package files,
+        if d in pkg.keys():
+            pkg[d] = True
+            subprocess.check_call(['mkdir', '-p',
+                os.path.join(src, d),
+            ])
+            for f in files:
+                if is_pkg_file(f):
+                    handle_pkg_file(os.path.join(subdir, f))
+                    subprocess.check_call(['cp',
+                        os.path.join(subdir, f),
+                        os.path.join(src, d, f),
+                    ])
+
+        # Copy test data.
+        if d in testdata.keys():
+            testdata[d] = True
+            subprocess.check_call(['cp', '-r',
+                subdir,
+                os.path.join(src, d),
+            ])
+
+        # Copy miscellaneous files.
+        for f in files:
+            k = os.path.join(d, f)
+            if k in misc.keys():
+                misc[k] = True
+                subprocess.check_call(['cp',
+                    os.path.join(subdir, f),
+                    os.path.join(src, d, f),
+                ])
+
+    for (d, v) in pkg.items():
+        if not v:
+            print(f'WARNING: package "{d}" not found in CIRCL repo', file=sys.stderr)
+
+    for (d, v) in testdata.items():
+        if not v:
+            print(f'WARNING: test data "{d}" not found in CIRCL repo', file=sys.stderr)
+
+    for (f, v) in misc.items():
+        if not v:
+            print(f'WARNING: file "{f}" not found in CIRCL repo', file=sys.stderr)
+
+    print(f"Creating README ...")
+    with open(os.path.join(src_circl, 'README'), 'w') as f:
+        readme = subprocess.check_output(['git', 'log',
+            '-n1',
+            '--format=fuller',
+        ], cwd=os.path.join(tmp, circl_name))
```

Comment:
To avoid the next line (`readme = str(readme, 'ascii')`) consider:
```suggestion
        ], cwd=os.path.join(tmp, circl_name), universal_newlines=True)
```

You are already using f-strings which require Python 3.6. If you can use Python 3.7, you can use `text=True` instead of `universal_newlines=True` for the same effect.

Reply:
Done.